src/app/view/components/_common/dot-dialog/dot-dialog.component.spec.ts

```diff
@@ -0,0 +1,186 @@
+            message: ''
+        });
+
+        fixture.detectChanges();
+        const confirm = de.query(By.css('p-confirmDialog'));
+        expect(confirm === null).toBe(false);
+    });
+
+    it('should have right attr in confirmation dialog', () => {
+        dialogService.confirm({
+            header: '',
+            message: ''
+        });
+
+        fixture.detectChanges();
+        const confirm = de.query(By.css('p-confirmDialog')).componentInstance;
+        expect(confirm.responsive).toBe(true, 'responsive');
+        expect(confirm.width).toBe('400', 'width');
+        expect(confirm.closable).toBe(false, 'closable');
+    });
+
+    it('should show dialog', () => {
+        dialogService.alert({
+            header: '',
+            message: ''
+        });
+
+        fixture.detectChanges();
+        const confirm = de.query(By.css('p-dialog'));
+        expect(confirm === null).toBe(false);
+    });
+
+    it('should have right attr in dialog', () => {
+        dialogService.alert({
+            header: 'Header Test',
+            message: ''
+        });
+
+        fixture.detectChanges();
+        const dialog = de.query(By.css('p-dialog')).componentInstance;
+
+        expect(dialog.closable).toBe(false, 'closable');
+        expect(dialog.header).toBe('Header Test', 'header');
+        expect(dialog.modal).toBe('modal', 'modal');
+        expect(dialog.responsive).toBe(true, 'responsive');
+        expect(dialog.visible).toBe(true, 'visible');
+        expect(dialog.width).toBe('400', 'width');
+    });
+
+    it('should add message to the dialog', () => {
+        dialogService.alert({
+            header: 'Header Test',
+            message: 'Hello world message'
+        });
+
+        fixture.detectChanges();
+        const message = de.query(By.css('.ui-dialog-content p'));
+        expect(message.nativeElement.textContent).toEqual('Hello world message');
+    });
+
+    it('should bind ok button in dialog', () => {
+        spyOn(dialogService, 'clearAlert');
+
+        dialogService.alert({
+            header: '',
+            message: '',
+        });
+
+        fixture.detectChanges();
+
+        const button = de.query(By.css('p-dialog button'));
+        button.nativeElement.click();
+        expect(dialogService.clearAlert).toHaveBeenCalledTimes(1);
+    });
+
+    it('should bind correctly to confirm dialog buttons', fakeAsync(() => {
+        spyOn(component, 'onClickConfirm');
+
+        dialogService.confirm({
+            header: '',
+            message: '',
+        });
+
+        fixture.detectChanges(); // ngIf
+        tick();
+        fixture.detectChanges(); // confirmation service make it happen
+
+        const buttons = de.queryAll(By.css('p-confirmDialog button'));
+        buttons[0].nativeElement.click();
+        expect(component.onClickConfirm).toHaveBeenCalledTimes(1);
+
+        buttons[1].nativeElement.click();
+        expect(component.onClickConfirm).toHaveBeenCalledTimes(2);
```

Comment:
I thin we shouldn't test thid, we should move the expect from the next test to here and remove the next test

Reply:
I see, I kinda like this way better so I make sure the buttons are binded to our click method but they also execute the primeng actions and I didn't want to check the whole thing in one test.